### ISSUE TITLE
Azure API profile latest does not define an entry for RecoveryServicesBackupClient

### ISSUE THREAD

**Issue by JeromeBollinger** [ansible/ansible_collections_azure#1370]
##### SUMMARY

When using the azure_rm_backuppolicy module to create a backuppolicy, it returns the following Object when the resource already exists: `{'changed': True, 'id': None, 'location': None, 'name': None, 'type': None, 'warnings': ['Azure API profile latest does not define an entry for RecoveryServicesBackupClient'], 'failed': False}`.
This only happens when the resource exists. Otherwise it will behave as expected.

Additionally I get this warning when using the module:
`[WARNING]: Azure API profile latest does not define an entry for RecoveryServicesBackupClient`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_backuppolicy 

##### ANSIBLE VERSION

```paste below
ansible [core 2.16.0]
  config file = [...]
  configured module search path = [...]
  ansible python module location = [...]
  ansible collection location = [...]
  executable location = [...]
  python version = 3.10.12 ([...]) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection         Version
------------------ -------
azure.azcollection 1.19.0 
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

Try to create a backuppolicy that already exists

```yaml
- name: Ensure BackupPolicy exists
  azure.azcollection.azure_rm_backuppolicy:
    vault_name: "test_vault"
    name: "TestBackupPolicy"
    resource_group: "test_rg"
    state: present
    backup_management_type: "AzureIaasVM"
    schedule_run_frequency: "Daily"
    schedule_run_time: "1"
    instant_recovery_snapshot_retention: 2
    daily_retention_count: 30
    time_zone: "UTC"
    api_profile: "latest"
  register: backup_policy_info
  delegate_to: 127.0.0.1

- name: Debug policy
  debug:
    msg: "{{ backup_policy_info }}"
```

##### EXPECTED RESULTS

Getting infos of the already existing backuppolicy

##### ACTUAL RESULTS

```
MSG:

{'changed': True, 'id': None, 'location': None, 'name': None, 'type': None, 'warnings': ['Azure API profile latest does not define an entry for RecoveryServicesBackupClient'], 'failed': False}
```

```paste below

```

**Comment by JeromeBollinger** [comment]
Current workaround is to just use the azure_rm_backuppolicy_info module

```
- name: Ensure CleanerBackupPolicy exists
  azure.azcollection.azure_rm_backuppolicy:
    vault_name: "test_vault"
    name: "TestBackupPolicy"
    resource_group: "test_rg"
    state: present
    backup_management_type: "AzureIaasVM"
    schedule_run_frequency: "Daily"
    schedule_run_time: "1"
    instant_recovery_snapshot_retention: 2
    daily_retention_count: 30
    time_zone: "UTC"
    api_profile: "latest"
  delegate_to: 127.0.0.1

- name: Get backup policy information
  azure.azcollection.azure_rm_backuppolicy_info:
    vault_name: "test_vault"
    name: "TestBackupPolicy"
    resource_group: "TestBackupPolicy"
  register: backup_policy
  delegate_to: 127.0.0.1

- name: Debug policy
  debug:
    msg: "{{ backup_policy }}"
```

**Comment by Fred-sun** [comment]
@JeromeBollinger Thank you for reporting this issue, it is a Bug, and the people who developed this module did not handle the idempotent line problem.

**Comment by Fred-sun** [comment]
Fixed by #1395
